datalake/src/main/java/com/sequenceiq/datalake/flow/dr/DatalakeDatabaseBackupActions.java

```diff
@@ -0,0 +1,173 @@
+package com.sequenceiq.datalake.flow.dr;
+
+import static com.sequenceiq.datalake.flow.dr.DatalakeDatabaseDrEvent.DATALAKE_DATABASE_BACKUP_FAILURE_HANDLED_EVENT;
+import static com.sequenceiq.datalake.flow.dr.DatalakeDatabaseDrEvent.DATALAKE_DATABASE_BACKUP_FINALIZED_EVENT;
+import static com.sequenceiq.datalake.flow.dr.DatalakeDatabaseDrEvent.DATALAKE_DATABASE_BACKUP_IN_PROGRESS_EVENT;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.StateContext;
+import org.springframework.statemachine.action.Action;
+
+import com.sequenceiq.datalake.entity.SdxDatabaseDrStatus;
+import com.sequenceiq.datalake.flow.SdxContext;
+import com.sequenceiq.datalake.flow.SdxEvent;
+import com.sequenceiq.datalake.flow.dr.event.DatalakeDatabaseBackupCouldNotStartEvent;
+import com.sequenceiq.datalake.flow.dr.event.DatalakeDatabaseBackupFailedEvent;
+import com.sequenceiq.datalake.flow.dr.event.DatalakeDatabaseBackupStartEvent;
+import com.sequenceiq.datalake.flow.dr.event.DatalakeDatabaseBackupSuccessEvent;
+import com.sequenceiq.datalake.flow.dr.event.DatalakeDatabaseBackupWaitRequest;
+import com.sequenceiq.datalake.service.AbstractSdxAction;
+import com.sequenceiq.datalake.service.sdx.dr.SdxDrService;
+import com.sequenceiq.flow.core.FlowEvent;
+import com.sequenceiq.flow.core.FlowParameters;
+import com.sequenceiq.flow.core.FlowState;
+
+@Configuration
+public class DatalakeDatabaseBackupActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatalakeDatabaseBackupActions.class);
+
+    private static final String OPERATION_ID = "OPERATION-ID";
+
+    @Inject
+    private SdxDrService sdxDrService;
+
+    @Bean(name = "DATALAKE_DATABASE_BACKUP_START_STATE")
+    public Action<?, ?> datalakeBackup() {
+        return new AbstractSdxAction<>(DatalakeDatabaseBackupStartEvent.class) {
+            @Override
+            protected SdxContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
+                    DatalakeDatabaseBackupStartEvent payload) {
+                return SdxContext.from(flowParameters, payload);
+            }
+
+            @Override
+            protected void prepareExecution(DatalakeDatabaseBackupStartEvent payload, Map<Object, Object> variables) {
+                super.prepareExecution(payload, variables);
+                variables.put(OPERATION_ID, payload.getDrStatus().getOperationId());
+            }
+
+            @Override
+            protected void doExecute(SdxContext context, DatalakeDatabaseBackupStartEvent payload, Map<Object, Object> variables) throws Exception {
```

Comment:
`throws Exception` is not necessary. same for all the `doExecute`

Reply:
will do